Sources/Sentry/SentryUIPerformanceTracker.m

```diff
@@ -0,0 +1,258 @@
+
+    SEL coderSelector = NSSelectorFromString(@"initWithCoder:");
+    SentrySwizzleInstanceMethod(UIViewController.class, coderSelector, SentrySWReturnType(id),
+        SentrySWArguments(NSCoder * coder), SentrySWReplacement({
+            [SentryUIPerformanceTracker swizzleViewControllerSubClass:[self class]];
+            return SentrySWCallOriginal(coder);
+        }),
+        SentrySwizzleModeOncePerClassAndSuperclasses, swizzleViewControllerInitWithCoder);
+
+    static const void *swizzleViewControllerInitWithNib = &swizzleViewControllerInitWithNib;
+    SEL nibSelector = NSSelectorFromString(@"initWithNibName:bundle:");
+    SentrySwizzleInstanceMethod(UIViewController.class, nibSelector, SentrySWReturnType(id),
+        SentrySWArguments(NSString * nibName, NSBundle * bundle), SentrySWReplacement({
+            [SentryUIPerformanceTracker swizzleViewControllerSubClass:[self class]];
+            return SentrySWCallOriginal(nibName, bundle);
+        }),
+        SentrySwizzleModeOncePerClassAndSuperclasses, swizzleViewControllerInitWithNib);
+#    pragma clang diagnostic pop
+#endif
+}
+
++ (void)swizzleViewControllerSubClass:(Class)class
+{
+#if SENTRY_HAS_UIKIT
+    // Swizzling only classes from the user app module to avoid track every UIKit view controller
+    // interaction.
+    static const char *appImage = nil;
+    if (appImage == nil) {
+        if ([UIApplication respondsToSelector:@selector(sharedApplication)]) {
+            UIApplication *app = [UIApplication performSelector:@selector(sharedApplication)];
+            appImage = class_getImageName(app.delegate.class);
+        }
+    }
+    if (appImage == nil || strcmp(appImage, class_getImageName(class)) != 0)
+        return;
+
+    // This are the five main functions related to UI creation in a view controller.
+    // We are swizzling it to track anything that happens inside one of this functions.
+    [SentryUIPerformanceTracker swizzleViewLayoutSubViews:class];
+    [SentryUIPerformanceTracker swizzleLoadView:class];
+    [SentryUIPerformanceTracker swizzleViewDidLoad:class];
+    [SentryUIPerformanceTracker swizzleViewWillAppear:class];
+    [SentryUIPerformanceTracker swizzleViewDidAppear:class];
+#endif
+}
+
++ (void)swizzleLoadView:(Class)class
+{
+#if SENTRY_HAS_UIKIT
+    // SentrySwizzleInstanceMethod declaration shadows a local variable. The swizzling is working
+    // fine and we accept this warning.
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wshadow"
+
+    SEL selector = NSSelectorFromString(@"loadView");
+    SentrySwizzleInstanceMethod(class, selector, SentrySWReturnType(void), SentrySWArguments(),
+        SentrySWReplacement({
+            NSString *name = [UIViewControllerHelper sanitizeViewControllerName:self];
+            NSString *spanId = [SentryPerformanceTracker.shared startSpanWithName:name
+                                                                        operation:@"ui.lifecycle"];
```

Comment:
Android and RN do navigation, https://github.com/getsentry/sentry-java/blob/main/sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java#L117-L118